SwanHelp/src/index.ts

```diff
@@ -1,25 +1,44 @@
 import { JupyterFrontEnd, JupyterFrontEndPlugin } from '@jupyterlab/application';
 import { ConfigWithDefaults, ConfigSection } from '@jupyterlab/services';
-import { IFrame, MainAreaWidget } from '@jupyterlab/apputils';
+import { Dialog, IFrame, MainAreaWidget, ICommandPalette } from '@jupyterlab/apputils';
 import { IMainMenu } from '@jupyterlab/mainmenu';
-import swanDialog from './dialog';
+import { ILauncher } from '@jupyterlab/launcher';
+import { showAboutDialog, showLoadingSpinnerDialog } from './dialog';
+import { swanGalleryIcon } from './icons';
+import { downloadUrlFromServer } from './gallery-download-helper';
+import { showDialog } from "@jupyterlab/apputils";
+
+function registerGalleryListener(app: JupyterFrontEnd, galleryUrl: string) {
+  window.addEventListener('message', async (event: any) => {
+    if (event.origin != galleryUrl) {
+      console.error(`SwanGallery: Failed to validate origin, message from ${event.origin} does not match ${galleryUrl} . Skipping downloading URL`);
+      return;
+    }
+    const dialog = showLoadingSpinnerDialog();
+    try {
+      const response = await downloadUrlFromServer(event.data);
+
+      app.commands.execute('filebrowser:open-path', {
+        path: response.path,
+        showBrowser: false
+      });
+      dialog.reject();
+    } catch (error) {
+      dialog.reject();
+      showDialog({ title: "Failed to download notebook", buttons: [Dialog.okButton()] });
+      console.error(
+        `The SwanGallery server extension appears to be missing.\n ${error}`
+      );
+    }
+  });
+}
 
 const defaultConfigs = {
   help: "https://swan.docs.cern.ch/",
   community: "https://cern.ch/swan-community",
   support: "https://cern.service-now.com/service-portal/function.do?name=swan",
-  gallery: "https://cern.ch/swan-gallery"
```

Comment:
Why don't we keep just 1 config and use this default here? If the extension is now called swan help, what's the reason behind spliting into 2 config files?

Reply:
Or does this make it easier somehow to configure from the k8s land?